docker/run.sh

```diff
@@ -87,37 +87,39 @@ parse_arguments() {
     done
 }
 
-# Set image and workspace variables
+# Set docker image used and workspace variables
 set_variables() {
-    # Check if map path is provided for default launch command
-    if [ "$MAP_PATH" == "" ] && [ "$LAUNCH_CMD" == "" ]; then
-        print_help
-        exit 1
-    fi
-
-    # Mount map path if provided
-    MAP="-v ${MAP_PATH}:/autoware_map:ro"
+    if [ "$option_devel" = "true" ]; then
+        # Set image based on option
+        IMAGE="ghcr.io/autowarefoundation/autoware:latest-devel"
 
-    # Set workspace path if provided and login with local user
-    if [ "$WORKSPACE_PATH" != "" ]; then
+        # Set workspace path, if not provided use the current directory
+        if [ "$WORKSPACE_PATH" = "" ]; then
+            WORKSPACE_PATH=$(pwd)
+        fi
         USER_ID="-e LOCAL_UID=$(id -u) -e LOCAL_GID=$(id -g) -e LOCAL_USER=$(id -un) -e LOCAL_GROUP=$(id -gn)"
         WORKSPACE="-v ${WORKSPACE_PATH}:/workspace"
-    fi
 
-    # Set default launch command if not provided
-    if [ "$LAUNCH_CMD" == "" ]; then
-        if [ "$WORKSPACE_PATH" != "" ]; then
-            LAUNCH_CMD="/bin/bash"
+        # Set launch command
+        LAUNCH_CMD="/bin/bash"
+    else
+        # Set image based on option
+        IMAGE="ghcr.io/autowarefoundation/autoware:latest-runtime"
+
+        # Set map path
+        if [ "$MAP_PATH" = "" ]; then
+            echo -e "\n------------------------------------------------------------"
+            echo -e "${RED}Note:${NC} The --map-path option is mandatory for the runtime. For development environment with shell access, use --devel option."
+            echo -e "------------------------------------------------------------"
+            exit 1
         else
-            LAUNCH_CMD=${DEFAULT_LAUNCH_CMD}
+            MAP="-v ${MAP_PATH}:/autoware_map:ro"
         fi
-    fi
 
-    # Set image based on option
-    if [ "$option_devel" == "true" ]; then
-        IMAGE="ghcr.io/autowarefoundation/autoware-openadk:latest-devel"
-    else
-        IMAGE="ghcr.io/autowarefoundation/autoware-openadk:latest-runtime"
+        # Set default launch command if not provided
+        if [ "$LAUNCH_CMD" = "" ]; then
+            LAUNCH_CMD=${DEFAULT_LAUNCH_CMD}
+        fi
```

Comment:
We can still make MAP_PATH and LAUNCH_CMD options valid when people are running devel image as well.